component/stream/stream.go

```diff
@@ -55,25 +54,38 @@ import (
 
 type Stream struct {
 	internal.BaseComponent
-	streamCache       *cache
-	streamOnly        bool // parameter used to check if its pure streaming
-	fileKeyLocks      *common.LockMap
-	cacheOnFileHandle bool // much better option for prefetching
+	cache      *cache
+	streamOnly bool // parameter used to check if its pure streaming
 }
 
 type StreamOptions struct {
 	BlockSize         uint64 `config:"block-size-mb" yaml:"block-size-mb,omitempty"`
-	BlocksPerFile     int    `config:"blocks-per-file" yaml:"blocks-per-file,omitempty"`
-	StreamCacheSize   uint64 `config:"cache-size-mb" yaml:"cache-size-mb,omitempty"`
+	BufferSizePerFile uint64 `config:"blocks-per-file" yaml:"blocks-per-file,omitempty"`
+	CacheSize         uint64 `config:"cache-size-mb" yaml:"cache-size-mb,omitempty"`
 	Policy            string `config:"policy" yaml:"policy,omitempty"`
-	CacheOnFileHandle bool   `config:"cache-on-file-handle" yaml:"cache-on-file-handle,omitempty"`
 	readOnly          bool   `config:"read-only"`
 	DiskPersistence   bool   `config:"disk-persistence"`
 	DiskPath          string `config:"disk-cache-path"`
 	DiskCacheSize     uint64 `config:"disk-size-mb"`
 	DiskTimeoutSec    uint64 `config:"disk-timeout-sec"`
 }
 
+type cache struct {
+	sync.RWMutex
+
+	evictionPolicy      common.EvictionPolicy
+	blocks              handlemap.LRUCache // blocks stored: {blockKey(off1, fileKey1): cacheBlock1, blockKey(off1, fileKey2): cacheBlock2, ...}
+	blockSize           int64
+	bufferSizePerHandle uint64 // maximum number of blocks allowed to be stored for a file
+	cacheSize           uint64 // maximum allowed configured number of blocks
+
+	diskBlocks      gcache.Cache // blocks stored on disk when persistence is on
```

Comment:
if we still use gcache then its eviction logic and locks will still kick in right ?